x-pack/platform/packages/shared/kbn-elastic-assistant/impl/assistant/api/conversations/use_fetch_current_user_conversations.ts

```diff
@@ -6,71 +6,177 @@
+  data: Record<string, Conversation>;
+  isLoading: boolean;
+  refetch: <TPageData>(
+    options?: RefetchOptions & RefetchQueryFilters<TPageData>
+  ) => Promise<QueryObserverResult<InfiniteData<FetchConversationsResponse>, unknown>>;
+  isFetched: boolean;
+  isFetching: boolean;
+  setPaginationObserver: (ref: HTMLDivElement) => void;
 }
 
-/**
- * API call for fetching assistant conversations for the current user
- *
- * @param {Object} options - The options object.
- * @param {HttpSetup} options.http - HttpSetup
- * @param {Function} [options.onFetch] - transformation function for conversations fetch result
- * @param {AbortSignal} [options.signal] - AbortSignal
- *
- * @returns {useQuery} hook for getting the status of the conversations
- */
 const query = {
   page: 1,
-  per_page: 99,
+  perPage: 28,
+  fields: ['id', 'title', 'apiConfig', 'updatedAt'],
 };
 
-export const CONVERSATIONS_QUERY_KEYS = [
-  ELASTIC_AI_ASSISTANT_CONVERSATIONS_URL_FIND,
-  query.page,
-  query.per_page,
-  API_VERSIONS.public.v1,
-];
+const formatFetchedData = (data: InfiniteData<FetchConversationsResponse> | undefined) =>
+  data?.pages.reduce((acc, curr) => {
+    return {
+      ...acc,
+      ...curr.data.reduce(
+        (conversationsArr, conversation) => ({
+          ...conversationsArr,
+          [conversation.id]: conversation,
+        }),
+        {}
+      ),
+    };
+  }, {});
 
+/**
+ * API call for fetching assistant conversations for the current user
+ */
 export const useFetchCurrentUserConversations = ({
   http,
-  onFetch,
+  fields = query.fields,
+  filter,
+  page = query.page,
+  perPage = query.perPage,
   signal,
+  sortField = 'updated_at',
+  sortOrder = 'desc',
   refetchOnWindowFocus = true,
   isAssistantEnabled,
-}: UseFetchCurrentUserConversationsParams) =>
-  useQuery(
-    CONVERSATIONS_QUERY_KEYS,
-    async () =>
-      http.fetch<FetchConversationsResponse>(ELASTIC_AI_ASSISTANT_CONVERSATIONS_URL_FIND, {
-        method: 'GET',
-        version: API_VERSIONS.public.v1,
-        query,
-        signal,
-      }),
-    {
-      select: (data) => onFetch(data),
-      keepPreviousData: true,
-      initialData: { ...query, total: 0, data: [] },
-      refetchOnWindowFocus,
-      enabled: isAssistantEnabled,
+  setTotalItemCount,
+}: UseFetchCurrentUserConversationsParams): FetchCurrentUserConversations => {
+  const queryFn = async ({ pageParam }: { pageParam?: UseFetchCurrentUserConversationsParams }) => {
+    return http.fetch<FetchConversationsResponse>(ELASTIC_AI_ASSISTANT_CONVERSATIONS_URL_FIND, {
+      method: 'GET',
+      version: API_VERSIONS.public.v1,
+      query: {
+        fields,
+        filter,
+        page: pageParam?.page ?? page,
+        per_page: pageParam?.perPage ?? perPage,
+        sort_field: sortField,
+        sort_order: sortOrder,
+      },
+      signal,
+    });
+  };
+
+  const getNextPageParam = (lastPage: FetchConversationsResponse) => {
```

Comment:
should memoize this

Reply:
I didn't think so because it does not take any props. But I can move it to outside of the hook as a helper

Reply:
you should still memoize it with an empty dependency array, otherwise on every render it will be recreated

Reply:
same with queryFn too